### PR TITLE
Fix `renderer-solid` not creating a reactive root

### LINKED ISSUE #847: 🐛 BUG: `renderer-solid` does not create a reactive root and cannot handle non native elements
### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### Describe the Bug

Currently the `renderer-solid` package only creates the native HTML element from Solid and inserts it in the DOM. This means that utilities such as `createEffect` and other reactivity between components is not working since it needs to be created within a [root](https://www.solidjs.com/docs/latest/api#createroot). The current implementation also does not allow for `<Dynamic>` to be the top level component since it is nor an HTML element nor an array of elements.

### Steps to Reproduce

1. `npm init astro --template framework-solid`
2. `npm add astro@next` to update to latest astro release
3. In the `Counter` component import `createEffect` from `solid-js` and try to log the value of `count()`
4. The effect will never run

### Link to Minimal Reproducible Example (Optional)

_No response_

### PR DESCRIPTION
**Pull request by pablo-abc**
## Changes

Fixes #847. Instead of creating the element and inserting it on the DOM, this PR makes use of Solid's `render` since it adds a reactive root to the component tree. Without it `createEffect` and other reactivity characteristics do not work. The `render` method also handles any kind of possible return from a Solid component since they are not necessarily native HTML elements.

## Testing

No tests were added, I couldn't find any tests regarding renderers to guide me in this case.

## Docs

Bug fix only

### CODE REVIEW

**Comment by FredKSchott** [comment]
Thanks for the PR! Assigning @natemoo-re to review

File: packages/renderers/renderer-solid/client.js [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-  // already attached. There's no VDOM, so there's no real need to "mount".
-  // Likewise, `children` can just reuse the nearest `astro-fragment` node.
-  const component = createComponent(Component, {
-    ...props,
-    children: element.querySelector('astro-fragment'),
-  });
+export default (element) => (Component, props, childHTML) => {
+  // Solid's `render` does not replace the element's children.
+  // Deleting the root's children is necessary before calling `render`.
+  element.replaceChildren();
```
Comment (matthewp) [inline packages/renderers/renderer-solid/client.js:7]: Won't this blow away state such as focused elements? Does Solid not support hydrating existing DOM?

Comment (matthewp) [reply packages/renderers/renderer-solid/client.js:7]: Should we be using this instead? https://www.solidjs.com/docs/1.0.0#hydrate

Comment (pablo-abc) [reply packages/renderers/renderer-solid/client.js:7]: Hey, @matthewp, thanks a lot for your time! This is similar to the previous implementation which was also destructive. Just using the `render` method to mount the root.

Solid _does_ support hydration but as far as I can tell it needs a script to be [generated and added to the head in the server](https://www.solidjs.com/guide#hydration-script). It does not throw any error if I generate it per component on `server.js` (but that'd be adding the same `<script>` tag over and over again per component. I see you can run a script before hydration using `hydrationPolyfills` or `polyfills` in `index.js` but in this case I'm not sure how important it is for this script tag to be _generated_.

A more important issue with hydration is that components that have children passed by Astro do not get hydrated at all. I'm guessing it has to do with [Solid not having generated those components](https://www.solidjs.com/guide#ssr-caveats). I've been testing locally and using `render` if children are passed and `hydrate` when children are not passed seems to work. But the most consistent solution seems to be to completely replace it with `render`.

Comment (pablo-abc) [reply packages/renderers/renderer-solid/client.js:7]: Scratch that last part, hydration failing like that seems to happen only when the top component is a fragment with more than one child element.

On another note, though. Using `hydrate` on a component that has a `Dynamic` somewhere within it fails completely.

These both seem to be issues within Solid, though. I'll mention them over there.

**Pull request merged**